mkosi/initrd.py

```diff
@@ -177,7 +178,7 @@ def main() -> None:
         )
 
         if args.output_dir:
-            with umask(~0o700):
+            with umask(~0o700) if os.getuid() == 0 else contextlib.nullcontext():  # type: ignore
```

Comment:
Very peculiar error, only in mypy, not in pyright this time
```
mkosi/initrd.py:181:18: error: "object" has no attribute "__enter__"  [attr-defined]
                with umask(~0o700) if os.getuid() == 0 else contextlib.nullcontext():
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
mkosi/initrd.py:181:18: error: "object" has no attribute "__exit__"  [attr-defined]
                with umask(~0o700) if os.getuid() == 0 else contextlib.nullcontext():
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Found 2 errors in 1 file (checked 51 source files)
```

Another fix instead of ignoring the type is
```suggestion
            with umask(~0o700) if os.getuid() == 0 else cast(mkosi.sandbox.umask, contextlib.nullcontext())
```
or the more verbose
```suggestion
            if os.getuid() == 0:
                with umask(~0o700):
                    Path(args.output_dir).mkdir(parents=True, exist_ok=True)
            else:
                Path(args.output_dir).mkdir(parents=True, exist_ok=True)
```

Reply:
Yes, very fancy issue... I buy the second, thanks!